app/services/product_drive_destroy_service.rb

```diff
@@ -1,64 +1,29 @@
 class ProductDriveDestroyService
-  attr_reader :product_drive, :user, :organization
-
-  def initialize(product_drive, user, organization)
-    @product_drive = product_drive
-    @user = user
-    @organization = organization
-  end
-
-  def self.call(product_drive, user, organization)
-    new(product_drive, user, organization).call
-  end
-
-  def self.can_destroy?(product_drive, user)
-    return false unless user.has_role?(Role::ORG_ADMIN, product_drive.organization)
-    product_drive.donations.empty?
-  end
-
-  def call
-    return unauthorized_error unless verify_role
-    return donation_error unless self.class.can_destroy?(product_drive, user)
-
-    product_drive.destroy
-
-    if product_drive.errors.any?
-      {
-        success: false,
-        message: product_drive.errors.full_messages.join("\n")
-      }
-    else
-      {
-        success: true,
-        message: "Product drive was successfully destroyed."
-      }
+  class << self
+    def call(product_drive, user, organization)
+      return Result.new(error: "You are not allowed to perform this action.") unless verify_role(user, organization)
+
+      unless can_destroy?(product_drive, user)
+        product_drive.errors.add(:base, "Cannot delete product drive with donations.")
+        raise ActiveRecord::RecordInvalid.new(product_drive)
```

Comment:
Why are we sometimes raising an error and sometimes setting the error to the Result object?

Reply:
I think I'd rather set the error to the Result object (not the ActiveRecord object).